File: app/src/main/java/de/roland_illig/phantomgo/CountResult.java
package de.roland_illig.phantomgo;

public class CountResult {
    public final int blackTerritory;
    public final int whiteTerritory;
    public final int blackCaptured;
    public final int whiteCaptured;
    public final int blackScore;
    public final int whiteScore;

    public CountResult(int blackTerritory, int whiteTerritory, int blackCaptured, int whiteCaptured) {
        this.blackTerritory = blackTerritory;
        this.whiteTerritory = whiteTerritory;
        this.blackCaptured = blackCaptured;
        this.whiteCaptured = whiteCaptured;
        this.blackScore = blackTerritory + blackCaptured;
        this.whiteScore = whiteTerritory + whiteCaptured;
    }

    @Override
    public String toString() {
        return "" +
                "black=" + blackTerritory + "+" + blackCaptured + ", " +
                "white=" + whiteTerritory + "+" + whiteCaptured;
    }
}

File: app/src/main/java/de/roland_illig/phantomgo/CountResult.kt
package de.roland_illig.phantomgo

class CountResult(
        val blackTerritory: Int,
        val whiteTerritory: Int,
        val blackCaptured: Int,
        val whiteCaptured: Int)
    : java.io.Serializable {
    val blackScore: Int = blackTerritory + blackCaptured
    val whiteScore: Int = whiteTerritory + whiteCaptured

    override fun toString(): String {
        return "" +
                "black=" + blackTerritory + "+" + blackCaptured + ", " +
                "white=" + whiteTerritory + "+" + whiteCaptured
    }
}
